src/device/usbd.c

```diff
@@ -484,39 +530,53 @@ static bool process_control_request(uint8_t rhport, tusb_control_request_t const
 
     //------------- Endpoint Request -------------//
     case TUSB_REQ_RCPT_ENDPOINT:
-      // Non standard request is not supported
-      TU_VERIFY( TUSB_REQ_TYPE_STANDARD == p_request->bmRequestType_bit.type );
+    {
+      bool drvHandled = false;
+      if(drvid != INVALID_DRV_TOKEN) {
+        usbd_control_set_complete_callback(usbd_class_drivers[drvid].control_complete);
 
-      switch ( p_request->bRequest )
-      {
-        case TUSB_REQ_GET_STATUS:
-        {
-          uint16_t status = usbd_edpt_stalled(rhport, tu_u16_low(p_request->wIndex)) ? 0x0001 : 0x0000;
-          tud_control_xfer(rhport, p_request, &status, 2);
-        }
-        break;
+        drvHandled = usbd_class_drivers[drvid].control_request(rhport, p_request);

```

Comment:
If the driver handles it, still let USBD core handle the request. I'm not sure if the class driver should handle the request before or after the core handles it.

Reply:
yeah I did something similar as well, though I forbid the class driver from stalling if it is standard request. Request can be stalled by returning false on either the `control_request() or control_complete()` callback